src/main/anovos/data_analyzer/quality_checker.py

```diff
@@ -9,19 +10,32 @@
 from anovos.shared.utils import attributeType_segregation, transpose_dataframe, get_dtype
 from pyspark.sql import functions as F
 from pyspark.sql import types as T
+from pyspark.sql import SparkSession, DataFrame
 
 
-def duplicate_detection(spark, idf, list_of_cols='all', drop_cols=[], treatment=False, print_impact=False):
+def _parse_columns(list_of_cols: Union[List[str], str]):
+    if isinstance(list_of_cols, str):
+        list_of_cols = [x.strip() for x in list_of_cols.split('|')]
+    return list(set(list_of_cols))
+
+
+def duplicate_detection(spark: SparkSession,
+                        input_df: DataFrame,
+                        columns: Union[List[str], str] = 'all',
+                        drop_columns: Optional[Union[List[str], str]] = None,
+                        *,
```

Comment:
Make the two flags explicit by forcing users to specify them as keyword arguments.